heed/src/env.rs

```diff
@@ -260,6 +261,57 @@ pub enum CompactionOption {
 }
 
 impl Env {
+    /// Return the size used by all the databases in the environment.
+    pub fn size(&self) -> Result<usize> {
+        let mut size = 0;
+
+        let mut stat = std::mem::MaybeUninit::uninit();
+        unsafe { mdb_result(ffi::mdb_env_stat(self.0.env, stat.as_mut_ptr())) }?;
+        let stat = unsafe { stat.assume_init() };
+
+        size += (stat.ms_leaf_pages + stat.ms_branch_pages + stat.ms_overflow_pages)
+            * stat.ms_psize as usize;
+
+        let rtxn = self.read_txn()?;
+
+        let dbi = self.raw_open_dbi(&rtxn, None, 0)?;
+
+        // we don’t want anyone to open an environment while we’re computing the stats
+        // thus we take a lock on all mutexes
+        let locks = (
+            OPENED_ENV.write().unwrap(),
+            self.0.dbi_open_mutex.lock().unwrap(),
+        );
+
+        // We’re going to iterate on the unnamed database
+        let mut cursor = RoCursor::new(&rtxn, dbi)?;
+
+        while let Some((key, _value)) = cursor.move_on_next()? {
+            if key.contains(&0) {
+                continue;
+            }
+
+            let key = String::from_utf8(key.to_vec()).unwrap();
+            let dbi = self.raw_open_dbi(&rtxn, Some(&key), 0);
```

Comment:
We could maybe change the `raw_open_dbi` function signature to directly accept a `&[u8]`, or not? What do you think? Instead of converting it into an UTF8 string.

Reply:
We could but in the `raw_open_database` and `raw_create_database`, we would then need to do this;
```rust
        let name = name.map(|n| CString::new(n).unwrap());
        let name_ptr = match name {
            Some(ref name) => name.as_bytes_with_nul().as_ptr() as *const _,
            None => ptr::null(),
        };
```

I think it’s easier to convert to utf-8 only in the stats 🤔